packages/web-components/src/select/select.spec.md

```diff
@@ -0,0 +1,100 @@
+# Select
+
+## Component Description
+
+The browser-native select control component allows users to choose one option from a dropdown list. It is commonly used in forms for selecting values from a predefined set of options.
+
+## Design Spec
+
+[Link to Design Spec in Figma](https://www.figma.com/file/0vOH481cv1VQyfQIX9JALn/Select?node-id=1319-163&t=nn2xgT4ieHjbkqQW-11)
+
+## Engineering Spec
+
+### Inputs
+
+- @attr appearance: "outline" "underline" "filled-darker" "filled-lighter" | "outline"
+- @attr size: "small" "medium" "large" | "medium"
+- @attr disabled: boolean
+- @attr multiple: boolean
+- @attr required: boolean
+- @attr name: string
+- @attr autofocus: boolean
+- @attr autocomplete: "on" "off" | "off"
+- @attr aria-label: string
+- @attr aria-labelledby: string
+- @attr aria-describedby: string
+
+### Outputs
+
+None
```

Comment:
I think you'll need to mimic the Select API here for setting/getting value as well as holding a reference to options, etc. Here's the broader API: https://developer.mozilla.org/en-US/docs/Web/API/HTMLSelectElement

Reply:
To make sure I'm tracking, would this look like:
```
get options() {
  return this.querySelectorAll('option');
}

get length() {
  return this.options.length;
}

get value() {
  const selectedOption = this.querySelector('option:checked');
  return selectedOption ? selectedOption.value : '';
}
...
etc
```

Reply:
I can't speak to the exact implementation here for now but I would avoid using querySelector in these ways. There are better methods at our disposal to hold a reference to these via observables + directives.

Reply:
Something like that, yeah. Some of the checks above aren't quite right. I might for instance guard against the *lack* of the control. `if (!this.control) { return }` and then you can just write code for each corresponding getter as though the element is there. As a note, I don't think you'd have "Default Slot Content" in the select. Recall, it supports option and optgroup elements only. The default value would need to be *an option itself* there.

Reply:
Gotcha, I'm tracking. And then for the HTML template I would do something like:

```
  <select
   ...
  >
    <slot name="option"></slot>
    <slot name="optgroup"></slot>
  </select>
```
And the usage would resemble: 
```
<fluent-select>
  <option slot="option" value="option1">Option 1</option>
  <option slot="option" value="option2">Option 2</option>
  <optgroup slot="optgroup" label="Group 1">
    <option value="option3">Option 3</option>
    <option value="option4">Option 4</option>
  </optgroup>
</fluent-select>
```

Reply:
I think these would both go in the default slot FWIW - the larger question here is if we want to support both. Then we should likely also ask whether we want to filter for those elements to ensure that other passed elements don't get rendered within it.

```html
<fluent-select>
  <option value="option1">Option 1</option>
  <option value="option2">Option 2</option>
  <optgroup label="Group 1">
    <option value="option3">Option 3</option>
    <option value="option4">Option 4</option>
  </optgroup>
</fluent-select>
```

Reply:
Would this be the first time we would filter out elements that don't align to the component? I can understand the reasoning why we want to do this, but given the relative lax guidelines in terms of letting developers perform non-standard operations on our components, do we want to lock this one down? It seems like it violates some of what we previously discussed about enforcing standards through linting vs. baked into the component.

Reply:
> the relative lax guidelines in terms of letting developers perform non-standard operations on our components, do we want to lock this one down

Can you elaborate on this? When it comes to something like menu, only `menuitem`, `menuitemradio`, and `menuitemcheckbox` elements are included in the taborder for instance. Because we support divider and in the future "groups", we don't filter...but there absolutely are rigid requirements on behaviors across the components.

I'm not sure that I'd agree with the term "lax" in the context of linting vs. baked here. Any proposals I've made on linting is around _design guidance_ where we want only subsets of appearances supported for certain components. The approach for FUIR9 has been to create "sub components" with only those variations supported. My recommendation is that runtime code being added to enforce design guidance is misled. I don't think that's the same as ensuring that only supported elements get *projected* into the slot of a component. The intent of those two things and the corresponding costs are different.

To be clear, I'm asking on filtering, not prescribing...though I think there is merit to it. What are the implications if we do vs. do not?

Reply:
After fooling around with this some more and putting some real code in place, I noticed the following behavior. I'm not sure if this is a deal breaker, but thought it might be worth discussing:

My template code looks like:
```
export function selectTemplate<T extends Select>(): ElementViewTemplate<T> {
  return html<T>`
    <template>
      <label part="label" for="control" ?hidden="${x => !x.label}"> ${x => x.label} </label>
      <select
        id="control"
        part="control"
        :autofocus="${x => x.autofocus}"
        :autocomplete="${x => x.autocomplete}"
        ?disabled="${x => x.disabled}"
        @change="${x => x.handleChange()}"
        @input="${x => x.handleInput()}"
      >
          <slot @slotchange="${x => x.handleSlotChange()}"></slot>
      </select>
    </template>
  `;
}
```

But when I try to pass the `option` values in when using it, they render outside of the `select` control in the Shadow DOM, ala:
![image](https://user-images.githubusercontent.com/49389/232609819-98612aaf-3d88-4b72-aa58-9281f37ee9de.png)

I might be able to use JavaScript to move those in but that seems expensive? Is there a way around this?

Reply:
> To be clear, I'm asking on filtering, not prescribing...though I think there is merit to it. What are the implications if we do vs. do not?

For me the preferred approach would be to have a runtime error in development mode (and ideally a lint rule) and nothing in production.
1. we do not want to break in production, filtered out elements are hard to debug.